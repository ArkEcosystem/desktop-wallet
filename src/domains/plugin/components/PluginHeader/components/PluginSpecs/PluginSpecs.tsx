import { Icon } from "app/components/Icon";
import React from "react";

type Props = {
	author: string;
	category: string;
	url: string;
	rating: string;
	version: string;
	size: string;
};

type GridColProps = {
	children: React.ReactNode;
	colSpan?: number;
	justify?: string;
	padding?: string;
};

type GridItemProps = {
	label: string;
	value: string;
	textDirection?: string;
	iconName?: string;
};

const GridItem = ({ label, value, textDirection }: GridItemProps) => (
	<div className={`flex flex-col ${textDirection && `text-${textDirection}`}`}>
		<span className="font-bold text-theme-neutral-400">{label}</span>
		<span className="font-bold text-theme-neutral-600">{value}</span>
	</div>
);

const GridCol = ({ children, colSpan, justify, padding }: GridColProps) => {
	const mountClassName = () => {
		let styles = "flex";

		if (colSpan) styles = `${styles} col-span-${colSpan}`;
		if (justify) styles = `${styles} justify-${justify}`;
		if (padding) styles = `${styles} ${padding}`;

		return styles;
	};

	return <div className={mountClassName()}>{children}</div>;
};

export const PluginSpecs = ({ author, category, url, rating, version, size }: Props) => (
	<div className="pt-5 mt-5 text-sm border-t border-dashed grid grid-cols-5 divide-x divide-theme-neutral-300 grid-flow-col border-theme-neutral-300">
		<GridCol>
			<div className="flex flex-col">
				<span className="font-bold text-theme-neutral-400">Author</span>
				<div className="flex items-center">
					<span className="font-bold text-theme-neutral-600">{author}</span>
					<div className="ml-3">
						<Icon name="OfficialArkPlugin" />
					</div>
				</div>
			</div>
		</GridCol>
		<GridCol padding="px-6">
			<GridItem label="Category" value={category} />
		</GridCol>
		<GridCol padding="px-6 -ml-5">
			<div className="flex flex-col">
				<span className="font-bold text-theme-neutral-400">URL</span>
				<span className="font-bold text-theme-primary">{url}</span>
			</div>
		</GridCol>
		<GridCol colSpan={2} justify="between" padding="px-6 -ml-8">
			<div className="flex flex-col">
				<span className="font-bold text-theme-neutral-400">Rating</span>
				<div className="flex items-center">
					<div className="pr-1 text-theme-warning-400">
						<Icon name="Star" />
					</div>
					<span className="font-bold text-theme-neutral-600">{rating}</span>
				</div>
			</div>
			<GridItem label="Version" value={`v.${version}`} textDirection="right" />
		</GridCol>
		<GridCol padding="pl-6">
			<GridItem label="Size" value={`${size} Mb`} textDirection="right" />
		</GridCol>
	</div>
);
