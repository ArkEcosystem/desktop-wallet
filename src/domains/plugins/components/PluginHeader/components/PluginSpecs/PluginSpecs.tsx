import React from "react";

type Props = {
	author: string;
	category: string;
	url: string;
	rating: string;
	version: string;
	size: string;
};

type GridColProps = {
	children: React.ReactNode;
	colSpan?: string;
	justify?: string;
	padding?: string;
};

type GridItemProps = {
	label: string;
	value: string;
	textDirection?: string;
};

const GridItem = ({ label, value, textDirection }: GridItemProps) => (
	<div className={`flex flex-col ${textDirection && `text-${textDirection}`}`}>
		<span className="font-bold text-theme-neutral-400">{label}</span>
		<span className="font-bold text-theme-neutral-600">{value}</span>
	</div>
);

const GridCol = ({ children, colSpan, justify, padding }: GridColProps) => {
	const mountClassName = () => {
		let styles = "flex";

		if (colSpan) styles = `${styles} col-span-${colSpan}`;
		if (justify) styles = `${styles} justify-${justify}`;
		if (padding) styles = `${styles} ${padding}`;

		return styles;
	};

	return <div className={mountClassName()}>{children}</div>;
};

export const PluginSpecs = ({ author, category, url, rating, version, size }: Props) => (
	<div className="grid grid-cols-5 divide-x divide-gray-400 text-sm mt-5 grid-flow-col border-t border-dashed border-theme-neutral pt-5">
		<GridCol>
			<GridItem label="Author" value={author} />
		</GridCol>
		<GridCol padding="px-8">
			<GridItem label="Category" value={category} />
		</GridCol>
		<GridCol padding="px-8">
			<GridItem label="URL" value={url} />
		</GridCol>
		<GridCol colSpan={2} justify="between" padding="px-8">
			<GridItem label="Rating" value={rating} />
			<GridItem label="Version" value={`v.${version}`} textDirection="right" />
		</GridCol>
		<GridCol padding="pl-8">
			<GridItem label="Size" value={`${size} Mb`} textDirection="right" />
		</GridCol>
	</div>
);
