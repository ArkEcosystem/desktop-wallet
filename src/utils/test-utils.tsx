import { render } from "@testing-library/react";
import { i18n } from "app/i18n";
import React from "react";
import { I18nextProvider } from "react-i18next";

const WithProviders = ({ children }: { children: React.ReactNode }) => {
	return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;
};

const customRender = (component: React.ReactElement, options: any) =>
	render(component, { wrapper: WithProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
