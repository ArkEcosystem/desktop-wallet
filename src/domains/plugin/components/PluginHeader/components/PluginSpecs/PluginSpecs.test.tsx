import React from "react";
import { render } from "test-utils";

import { PluginSpecs } from "./PluginSpecs";

describe("PluginSpecs", () => {
	it("should render properly", async () => {
		const { asFragment, findByText } = render(
			<PluginSpecs
				author="ARK Ecosystem"
				category="Utility"
				url="github.com"
				rating="4.6"
				version="1.3.8"
				size="4.2"
			/>,
		);

		expect(await findByText("ARK Ecosystem")).toBeTruthy();
		expect(await findByText("Utility")).toBeTruthy();
		expect(await findByText("github.com")).toBeTruthy();
		expect(await findByText("4.6")).toBeTruthy();
		expect(await findByText("v.1.3.8")).toBeTruthy();
		expect(await findByText("4.2 Mb")).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});
});
