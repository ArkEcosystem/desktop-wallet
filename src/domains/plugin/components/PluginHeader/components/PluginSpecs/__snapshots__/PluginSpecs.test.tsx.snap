// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginSpecs should render properly 1`] = `
<DocumentFragment>
  <div
    class="pt-5 mt-5 text-sm border-t border-dashed grid grid-cols-5 divide-x divide-theme-neutral-300 grid-flow-col border-theme-neutral-300"
  >
    <div
      class="flex"
    >
      <div
        class="flex flex-col"
      >
        <span
          class="font-bold text-theme-neutral-400"
        >
          Author
        </span>
        <div
          class="flex items-center"
        >
          <span
            class="font-bold text-theme-neutral-600"
          >
            ARK Ecosystem
          </span>
          <div
            class="ml-3"
          >
            <div
              class="sc-AxjAm cpczEI"
              height="1em"
              width="1em"
            >
              <svg>
                official-ark-plugin.svg
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex px-6"
    >
      <div
        class="flex flex-col undefined"
      >
        <span
          class="font-bold text-theme-neutral-400"
        >
          Category
        </span>
        <span
          class="font-bold text-theme-neutral-600"
        >
          Utility
        </span>
      </div>
    </div>
    <div
      class="flex px-6 -ml-5"
    >
      <div
        class="flex flex-col"
      >
        <span
          class="font-bold text-theme-neutral-400"
        >
          URL
        </span>
        <span
          class="font-bold text-theme-primary"
        >
          github.com
        </span>
      </div>
    </div>
    <div
      class="flex col-span-2 justify-between px-6 -ml-8"
    >
      <div
        class="flex flex-col"
      >
        <span
          class="font-bold text-theme-neutral-400"
        >
          Rating
        </span>
        <div
          class="flex items-center"
        >
          <div
            class="pr-1 text-theme-warning-400"
          >
            <div
              class="sc-AxjAm cpczEI"
              height="1em"
              width="1em"
            >
              <svg>
                star.svg
              </svg>
            </div>
          </div>
          <span
            class="font-bold text-theme-neutral-600"
          >
            4.6
          </span>
        </div>
      </div>
      <div
        class="flex flex-col text-right"
      >
        <span
          class="font-bold text-theme-neutral-400"
        >
          Version
        </span>
        <span
          class="font-bold text-theme-neutral-600"
        >
          v.1.3.8
        </span>
      </div>
    </div>
    <div
      class="flex pl-6"
    >
      <div
        class="flex flex-col text-right"
      >
        <span
          class="font-bold text-theme-neutral-400"
        >
          Size
        </span>
        <span
          class="font-bold text-theme-neutral-600"
        >
          4.2 Mb
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
